test/lib/run-dev.test.js

```diff
@@ -0,0 +1,1015 @@
+} = require('../../src/lib/run-dev')
+
+jest.useFakeTimers()
+
+jest.mock('connect-livereload')
+jest.mock('node:path')
+jest.mock('fs-extra')
+jest.mock('get-port')
+
+jest.mock('livereload', () => {
+  return {
+    createServer: jest.fn(() => {
+      return {
+        watch: jest.fn(),
+        refresh: jest.fn(),
+        server: {
+          once: jest.fn((_, fn) => {
+            fn() // call right away, coverage
+            jest.runOnlyPendingTimers()
+          })
+        }
+      }
+    })
+  }
+})
+
+jest.mock('node:https', () => {
+  return {
+    createServer: jest.fn(() => {
+      return {
+        listen: jest.fn((_, fn) => {
+          fn() // call right away, coverage
+        }),
+        close: jest.fn()
+      }
+    })
+  }
+})
+
+// unmock to test proper returned urls from getActionUrls
+jest.unmock('@adobe/aio-lib-runtime')
+
+// create a Response object
+const createRes = ({ mockStatus, mockSend, mockSet = jest.fn() }) => {
+  const obj = {
+    set: mockSet,
+    status: mockStatus,
+    send: mockSend
+  }
+  mockSet.mockReturnValue(obj)
+  mockStatus.mockReturnValue(obj)
+  mockSend.mockReturnValue(obj)
+
+  return obj
+}
+
+// create a Request object
+const createReq = ({ url, body, headers = [], query, method = 'GET', is = jest.fn() }) => {
+  return {
+    body,
+    headers,
+    query,
+    method,
+    params: [url],
+    is
+  }
+}
+
+const createConfig = ({ distDev = 'dist', hasFrontend, hasBackend, packageName = 'mypackage', actions = {}, sequences = {} }) => {
+  return {
+    web: {
+      distDev
+    },
+    ow: {
+      namespace: 'mynamespace',
+      auth: 'myauthkey',
+      defaultApihost: 'https://localhost',
+      apihost: 'https://localhost'
+    },
+    app: {
+      hostname: 'https://localhost',
+      defaultHostname: 'https://adobeio-static.net',
+      hasFrontend,
+      hasBackend
+    },
+    manifest: {
+      full: {
+        packages: {
+          [packageName]: {
+            actions,
+            sequences
+          }
+        }
+      }
+    }
+  }
+}
+
+const createRunOptions = ({ cert, key }) => {
+  return {
+    parcel: {
+      https: {
+        cert,
+        key
+      }
+    }
+  }
+}
+
+const createBundlerEvent = ({ type = 'buildSuccess', diagnostics = 'some diagnostics', changedAssets = [], bundles = [] } = {}) => {
+  return {
+    type,
+    diagnostics,
+    changedAssets: new Map(changedAssets), // param is a key value array [[key, value][key, value]]
+    bundleGraph: {
+      getBundles: jest.fn(() => bundles)
+    }
+  }
+}
+
+beforeEach(() => {
+  mockLogger.mockReset()
+  mockGetPort.mockReset()
+  mockExpress.mockReset()
+})
+
+test('exports', () => {
+  expect(runDev).toBeDefined()
+  expect(serveWebAction).toBeDefined()
+  expect(serveNonWebAction).toBeDefined()
+  expect(httpStatusResponse).toBeDefined()
+  expect(invokeAction).toBeDefined()
+  expect(invokeSequence).toBeDefined()
+  expect(isRawWebAction).toBeDefined()
+  expect(isWebAction).toBeDefined()
```

Comment:
missing some exports

Reply:
fixed in https://github.com/adobe/aio-cli-plugin-app-dev/pull/25/commits/808fbbf18a6e071d300bf25a8a4f5e923a919332